src/app/child-dev-project/attendance/add-day-attendance/roll-call-setup/roll-call-setup.component.ts

```diff
@@ -54,21 +53,21 @@ export class RollCallSetupComponent implements OnInit {
     );
 
     for (const activity of this.visibleActivities) {
-      const newEvent = this.createEventForActivity(activity);
+      const newEvent = await this.createEventForActivity(activity);
       if (newEvent) {
-        this.existingEvents.push(newEvent);
+        this.existingEvents.push(await newEvent);
       }
     }
   }
 
-  showMore() {
+  async showMore() {
     const additionalActivities = this.allActivities.filter(
       (a) => !this.visibleActivities.includes(a)
     );
     for (const activity of additionalActivities) {
       const newEvent = this.createEventForActivity(activity);
       if (newEvent) {
-        this.existingEvents.push(newEvent);
+        this.existingEvents.push(await newEvent);
```

Comment:
the `await` should probably go above - `Promise<undefined>` will also pass `if (newEvent)`